File: backend_java/src/main/java/com/CafeManagement/service/OrderService.java
package com.CafeManagement.service;

import java.util.ArrayList;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import com.CafeManagement.dto.OrderRequest;
import com.CafeManagement.dto.OrderResponse;
import com.CafeManagement.model.Order;
import com.CafeManagement.repo.OrderRepo;

@Service
@Transactional
public class OrderService {
	OrderRepo repo;

	@Autowired
	public OrderService(OrderRepo repo) {
		this.repo = repo;
	}

	public void CreateOrder(OrderRequest req, String bill_id) throws Exception {
		repo.CreateOrder(req, bill_id);
	}

	public List<OrderResponse> GetAllOrder() throws Exception {
		List<Order> orders = repo.GetAllOrders();
		List<OrderResponse> responses = new ArrayList<OrderResponse>();
		for (Order order : orders) {
			responses.add(new OrderResponse(order));
		}

		return responses;
	}

	public List<OrderResponse> GetOrderByBillId(String id) throws Exception {
		List<Order> orders = repo.GetOrdersByBillId(id);
		List<OrderResponse> responses = new ArrayList<OrderResponse>();
		for (Order order : orders) {
			responses.add(new OrderResponse(order));
		}
		return responses;
	}

	// public void UpdateOrderById(OrderRequest req, int order_id, String bill_id)
	// throws Exception {
	// repo.UpdateOrderById(req, bill_id, order_id);
	// }

	public void DeleteAllOrder() throws Exception {
		repo.DeleteAllOrders();
	}

}

File: backend_go/internal/service/order_service.go
package service

import (
	"context"

	"github.com/HeHHeyboi/Cafe_Management/backend/internal/dto"
	"github.com/HeHHeyboi/Cafe_Management/backend/internal/model"
	"github.com/HeHHeyboi/Cafe_Management/backend/internal/repository"
)

type OrderService struct {
	repo repository.OrderRepo
}

func NewOrderService(repo repository.OrderRepo) OrderService {
	return OrderService{repo}
}

func (o OrderService) CreateOrder(ctx context.Context, billId string, req dto.OrderRequest) error {
	order := model.Order{
		Size:   req.Size,
		Type:   req.Type,
		MenuId: req.MenuID,
		Amount: req.Amount,
	}
	err := o.repo.CreateNewOrder(ctx, billId, order)
	if err != nil {
		return err
	}

	return nil
}

func (o OrderService) GetAllOrder(ctx context.Context) ([]dto.OrderResponse, error) {
	orders, err := o.repo.GetAllOrder(ctx)
	if err != nil {
		return nil, err
	}

	var response []dto.OrderResponse
	for _, order := range orders {
		res := dto.OrderResponse{
			Size:   order.Size,
			Type:   order.Type,
			Id:     order.Id,
			MenuId: order.MenuId,
			Amount: order.Amount,
			Total:  order.Total,
		}
		response = append(response, res)
	}

	return response, nil
}

func (o OrderService) GetOrdersByBillID(ctx context.Context, billID string) ([]dto.OrderResponse, error) {
	orders, err := o.repo.GetOrderFromBill(ctx, billID)
	if err != nil {
		return nil, err
	}

	var response []dto.OrderResponse
	for _, order := range orders {
		res := dto.OrderResponse{
			Size:   order.Size,
			Type:   order.Type,
			Id:     order.Id,
			MenuId: order.MenuId,
			Amount: order.Amount,
			Total:  order.Total,
		}
		response = append(response, res)
	}

	return response, nil
}

func (o OrderService) DeleteAllOrder(ctx context.Context) error {
	err := o.repo.DeleteAllOrder(ctx)
	if err != nil {
		return err
	}
	return nil
}
